test-suite/test-sources.sh

```diff
@@ -0,0 +1,262 @@
+#!/bin/sh
+#
+## Copyright (C) 1996-2022 The Squid Software Foundation and contributors
+##
+## Squid software is distributed under GPLv2+ license and includes
+## contributions from numerous individuals and organizations.
+## Please see the COPYING and CONTRIBUTORS files for details.
+##
+
+# This script checks source code compliance with a few Squid Project policies
+# and recommendations. Some of the checks are applied to a subset of
+# repository files, depending on the environment variables (first match wins):
+# * If $PULL_REQUEST_NUMBER is set, checks sources modified by that Github PR;
+# * If $FORK_POINT is set, checks sources modified since that commit;
+# * Otherwise, checks sources modified since the parent commit (HEAD^).
+STARTING_POINT="HEAD^"
+if test -n "${PULL_REQUEST_NUMBER}"
+then
+    STARTING_POINT="refs/pull/${PULL_REQUEST_NUMBER}/merge^1"
+elif test -n "${FORK_POINT}"
+then
+    STARTING_POINT="${FORK_POINT}"
+fi
+
+# Customizable check outcome description logged by run_one_check()
+CHECK_OUTCOME_PHRASE=""
+
+# XXX: echo_*() and run() functions are duplicated in test-functionality.sh.
```

Comment:
We can remove this code duplication if we move duplicated functions into a third script that the two posted scripts will include by "sourcing" it. I am not sure whether the increased complexity is worth it, but am ready to implement that fix if others think it is a good idea. The code duplication will only become worse with time, of course.